<!--REQUEST_ID=11a30a70-a844-48fe-9732-505e0dd825f6-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.50362&attempt_id=d87a8ecb-9d4a-4b48-9d66-8e47c14835ee)

Attempted: x265

<details><summary>Partial log (click to expand)</summary><p>

```
[ 96%] Building CXX object CMakeFiles/cli.dir/output/yuv.cpp.o
[ 98%] Building CXX object CMakeFiles/cli.dir/x265.cpp.o
[100%] Linking CXX executable x265
libx265.so.165: undefined reference to `x265::detect512()'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/cli.dir/build.make:205: x265] Error 1
make[1]: *** [CMakeFiles/Makefile2:213: CMakeFiles/cli.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
builder for '/nix/store/ikwmknn9m66yqmgwafh7lkjrlcdi64s9-x265-2.9.drv' failed with exit code 2
error: build of '/nix/store/ikwmknn9m66yqmgwafh7lkjrlcdi64s9-x265-2.9.drv' failed
```
</p></details>

--- a/pkgs/development/libraries/x265/default.nix
+++ b/pkgs/development/libraries/x265/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, cmake, yasm
+{ stdenv, fetchurl, fetchpatch, cmake, yasm
 , debugSupport ? false # Run-time sanity checks (debugging)
 , highbitdepthSupport ? false # false=8bits per channel, true=10/12bits per channel
 , werrorSupport ? false # Warnings as errors
@@ -16,19 +16,28 @@ in
 
 stdenv.mkDerivation rec {
   name = "x265-${version}";
-  version = "2.7";
+  version = "2.9";
 
   src = fetchurl {
     urls = [
-      "http://get.videolan.org/x265/x265_${version}.tar.gz"
-      "https://github.com/videolan/x265/archive/${version}.tar.gz"
+      "https://get.videolan.org/x265/x265_${version}.tar.gz"
+      "ftp://ftp.videolan.org/pub/videolan/x265/x265_${version}.tar.gz"
     ];
-    sha256 = "18llni1m8kfvdwy5bp950z6gyd0nijmvi3hzd6gd8vpy5yk5zrym";
+    sha256 = "090hp4216isis8q5gb7bwzia8rfyzni54z21jnwm97x3hiy6ibpb";
   };
 
   enableParallelBuilding = true;
 
-  patchPhase = ''
+  patches = [
+    # Fix issue #442 (linking issue on non-x86 platforms)
+    # Applies on v2.9 only, this should be removed at next update
+    (fetchpatch {
+      url = "https://bitbucket.org/multicoreware/x265/commits/471726d3a0462739ff8e3518eb1a1e8a01de4e8d/raw";
+      sha256 = "0mj8lb8ng8lrhzjavap06vjhqf6j0r3sn76c6rhs3012f86lv928";
+    })
+  ];
+
+  postPatch = ''
     sed -i 's/unknown/${version}/g' source/cmake/version.cmake
   '';
 

